sourcegraph/sourcegraph
# admin-analytics: make "analytics" and "user management" filter state synced with URL query parameters

Closes https://github.com/sourcegraph/sourcegraph/issues/40439.

This PR:
* Updates "analytics" and "user management" filter state to be synced with URL query parameters
* Fixes user-management reloading of legends when action end
* Adds "useURLSyncState" reusable hook + unit tests

## Test plan
- `sg start`
- Open http://localhost:3080/site-admin/users?searchText=&limit=25&lastActivePeriod=THIS_MONTH&orderBy=USERNAME&descending=true
- Check that all chart and table filters in analytics and user management pages are synced to and from URL


## App preview:

- [Web](https://sg-web-erzhtor-sync-admin-analytics.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-zvmxhicthw.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
